### PR TITLE
Improve FE/BE deployment to AWS - avoid downtime while deploying

### PR DESCRIPTION
**Pull request by ruslanpalagin**
https://app.clickup.com/t/860pm1xg9

### CODE REVIEW

**Review by dmitriy-green — DISMISSED** [review]

**Comment by dmitriy-green** [comment]
Please fix conflicts

File: api/infra/docker.js [hunk 1]
```diff
@@ -40,8 +40,27 @@ async function deployStagingApi(imageTag){
...
+// subEnv = 'blue' | 'green
+async function deployProductionApi(imageTag, subEnv){
+  const envFile = subEnv === 'blue' ? '.i1_blue_api_env' : '.i1_green_api_env'
+  const serverName = subEnv === 'blue' ? 'i1_blue_api' : 'i1_green_api'
+  const icon = subEnv === 'blue' ? ':large_blue_circle:' : ':large_green_circle:'
+
+  const sign = await getSign(imageTag)
+  const cmd = getDockerRunCmdScalableVersion(config, imageTag, envFile)
+  try {
+    await notify(`🙌 Production Api ${icon}: deploying... (${sign})`,'#dev-deploys')
```
Comment (boivlad) [inline api/infra/docker.js:52]: May add commit hash to message?

File: api/src/app.controller.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
-  constructor(private readonly appService: AppService) {}
+  constructor(
+    private readonly appService: AppService,
+    private readonly config: ConfigService,
+  ) {}
 
   @Get()
   getHello(): string {
-    return this.appService.getHello();
+    return 'SE:' + this.config.get('SUB_ENV');
```
Comment (boivlad) [inline api/src/app.controller.ts:14]: Are you sure this should be here?

**Comment by ruslanpalagin** [comment]
resolved

**Review by boivlad — APPROVED** [review]

File: api/src/app.controller.ts (hunk 2 above)
Comment (ruslanpalagin) [reply api/src/app.controller.ts:14]: yep, just for debug I think may be very helpful

**Pull request merged**